modules/project-comments/client/config/commentperiod.routes.js

```diff
@@ -236,26 +259,25 @@ angular.module('comment').config(['$stateProvider', 'moment', "_", function ($st
 					$scope.hasErrors = true;
 					$scope.errorMessage = 'Post, Vet and Classify Comments roles are all required.  See Roles & Permissions tab.';
 				} else {
+					$scope.busy = true;
 
 					CommentPeriodModel.save($scope.period)
 					.then(function (model) {
-						// console.log ('period was saved',model);
-						// save the comments so that we pick up the (potential) changes to the period permissions...
-						return CommentModel.getAllCommentsForPeriod(model._id);
-					})
-					.then(function (comments) {
-						Promise.resolve()
-						.then(function () {
-							return comments.reduce(function (current, value, index) {
-								return CommentModel.save(value);
-							}, Promise.resolve());
-						});
+						if (!rolesChanged(model)) {
+							return [];
```

Comment:
Not sure if you really need to return an empty array here. The following code doesn't do anything with the returned result.